### ISSUE TITLE
Unable to decode ICO images?

### ISSUE THREAD

**Issue by vthommeret** [mat/besticon#8]
I'm using your library to fetch the icon for a given website and convert it to a PNG for upload. This generally works for JPG, GIF, PNG, etc... images, but not ICO.

Calling `ico.ParseIco` seems to return reasonable metadata, e.g.:

`github.ico: MS Windows icon resource - 2 icons, 32x32, 256-colors`

But if I try and call `image.Decode`, I always get back "unknown format". I've read that there's not a reliable magic string for ICO, so I've tried calling `ico.DecodeImage` directly like this:

https://gist.github.com/vthommeret/d74d15549306571c83ca969d2f56c116

And I get this output:

```
github.ico: len(bytes) 5174
err: image: unknown format
format: 
Failed to parse github.ico as icon file
<nil>
```

I've verified that the "ico" package is being imported and the format is registered. Am I missing something, or is there a way get an `image.Image` from an ICO file? The `icoparser` is just an example of getting the metadata.

Thanks!

**Comment by mat** [comment]
I am sorry but unfortunately you're not missing something, it just doesn't work ;-) `decodeImage` is not at all a working implementation.

(When I started this there was simply no `ico` parser package in go that I could use so I had to roll my own and I've only gotten around to implement a good enough implementation of the `DecodeConfig` function which is enough to figure out the image dimensions.)

By the way this is also the reason why (a) websites having only `.ico` icons will only return a grey image and (b) I never promoted this ico package to become a separate GitHub project.

Contributions or pointers to full image/ico packages welcome.
